httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/H2OnlyClientProtocolNegotiator.java

```diff
@@ -131,7 +134,10 @@ public void outputReady(final IOSession session) throws IOException {
     @Override
     public void inputReady(final IOSession session, final ByteBuffer src) throws IOException {
         if (src != null) {
-            throw new ProtocolNegotiationException("Unexpected input");
+            if (inBuf == null) {
+                inBuf = BufferedData.allocate(src.remaining());
+            }
+            inBuf.put(src);
```

Comment:
I think we need to handle a potential buffer overflow here in the case an existing `inBuf` doesn't have enough room for `src`